File: FINAL/graph1.py
import matplotlib.pyplot as plt

# open file and read line starting from line 6 (before are all comments)
inputFile = open('20150801.as2types.txt', 'r')
readFile = inputFile.readlines()
readFile = readFile[6:]
data = []

for line in readFile:
    field = line.split('|')
    field[-1] = field[-1].rstrip()
    data.append(field)

# initialize count
tCount = 0
cCount = 0
eCount = 0
totalCount = 0

# looking for matching fields
for x in data:
    if x[2] == 'Transit/Access':
        tCount += 1
    elif x[2] == 'Content':
        cCount += 1
    elif x[2] == 'Enterpise':
        eCount += 1
    totalCount += 1

    # Calculate percentage for enterprise
    tPer = tCount/totalCount;
    # Calculate percentage for transit/access
    cPer = cCount/totalCount;
    # Calculate percentage for content
    ePer = eCount/totalCount;

    dataArray = [tPer, cPer, ePer]

labels = ['Transit/Access', 'Content', 'Enterprise']
plt.pie(dataArray, labels=labels, autopct = "%1.3f%%")
plt.title('Graph 1 - % of AS Distribution')
plt.show()

inputFile.close()

File: FINAL/graph1.m
clc
close all
clear 

file=fopen('20150801as2types.txt');
Dataset=textscan(file,'%d %s %s');
fclose(file);

sequence=Dataset{1};
model=Dataset{3};

T='Transit';
C='Content';
E='Enterpise';

Transit=0;
Content=0;
Enterpise=0;

for i=1:51507
    t=strcmp(model(i),T);
    c=strcmp(model(i),C);
    e=strcmp(model(i),E);
    if t==1
        Transit=Transit+1;
    elseif c==1
        Content=Content+1;
    elseif e==1
        Enterpise=Enterpise+1;
    end
end

data=[Transit Content Enterpise];
label={'Transit ASes','Content ASes','Enterpise ASes'};

bili=data/sum(data);
baifenbi=num2str(bili'*100,'%1.2f');
baifenbi=[repmat(blanks(2),length(data),1),baifenbi,repmat('%',length(data),1)];
baifenbi=cellstr(baifenbi);

Label=strcat(label,baifenbi');
pie(data,Label)

